### PR TITLE
Binary Version Updates

### LINKED ISSUE #843: Include Build ID in Binary Version
### Describe the feature you'd like supported

When built by AZP, we should encode the build ID into the version number. 1.0.0.109915 for instance. Also, when built by the public CI, we should probably tag it as such.

### PR DESCRIPTION
**Pull request by nibanks**
Fixes #843. Updates the version number to include the AZP build ID.

### CODE REVIEW

File: src/inc/msquic.ver [hunk 1]
```diff
@@ -3,21 +3,27 @@
...
 #define VER_INTERNALNAME_STR        "msquic"
 #define VER_LEGALCOPYRIGHT_STR      "\251 Microsoft Corporation."
 #define VER_PRODUCTNAME_STR         "Microsoft\256 QUIC"
-
-#define VER_FILEVERSION             1,0,0,0
-#define VER_FILEVERSION_STR         "1.0.0\0"
-
-#define VER_PRODUCTVERSION          1,0,0,0
-#define VER_PRODUCTVERSION_STR      "1.0.0\0"
+#define VER_PRODUCTVERSION_STR      "1.0.0." STR(VER_BUILD_ID) VER_SUFFIX "\0"
```
Comment (ThadHouse) [inline src/inc/msquic.ver:22]: Is changing this going to break the version update and packaging scripts?

Comment (nibanks) [reply src/inc/msquic.ver:22]: Perhaps. I'll look into it.

Comment (nibanks) [reply src/inc/msquic.ver:22]: Fixed the update version script. I think that should be it.

Comment (ThadHouse) [reply src/inc/msquic.ver:22]: create-release.ps1 looks to need the same line changed I think.

**Review by ThadHouse — APPROVED** [review]

File: src/inc/msquic.ver [hunk 2]
```diff
@@ -3,21 +3,28 @@
...
+#endif
+
 #define VER_COMPANYNAME_STR         "Microsoft Corporation"
 #define VER_FILEDESCRIPTION_STR     "Microsoft\256 QUIC Library"
 #define VER_INTERNALNAME_STR        "msquic"
 #define VER_LEGALCOPYRIGHT_STR      "\251 Microsoft Corporation."
 #define VER_PRODUCTNAME_STR         "Microsoft\256 QUIC"
-
+#define VER_PRODUCTVERSION_STR      "1.0.0." STR(VER_BUILD_ID) STR(VER_SUFFIX) "\0"
+#define VER_FILEVERSION_STR         "1.0.0." STR(VER_BUILD_ID) "\0"
```
Comment (nibanks) [inline src/inc/msquic.ver:23]: Make this back to zero to match the other file version.

**Pull request merged**